src/tests/banners/bannerSelection.tsx

```diff
@@ -34,70 +49,150 @@ const fetchBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date>
-    if (targeting.engagementBannerLastClosedAt) {
-        const closedAt = targeting.engagementBannerLastClosedAt;
-        const region = readerRevenueRegionFromCountryCode(targeting.countryCode);
-
-        const [, getCached] = caches[region];
+export const redeployedSinceLastClosed = (
+    targeting: BannerTargeting,
+    bannerType: BannerType,
+): Promise<boolean> => {
+    const { subscriptionsBannerLastClosedAt, engagementBannerLastClosedAt } = targeting;
 
-        return getCached().then(deployDate => deployDate > new Date(closedAt));
-    } else {
+    if (
+        (bannerType === 'subscriptions' &&
+            typeof subscriptionsBannerLastClosedAt === 'undefined') ||
+        (bannerType === 'contributions' && typeof engagementBannerLastClosedAt === 'undefined')
+    ) {
         return Promise.resolve(true);
     }
+
+    const region = readerRevenueRegionFromCountryCode(targeting.countryCode);
+
+    if (bannerType === 'subscriptions') {
+        const [, getCached] = caches.subscriptions[region];
+        return getCached().then(deployDate => {
+            return (
+                !subscriptionsBannerLastClosedAt ||
+                deployDate > new Date(subscriptionsBannerLastClosedAt)
+            );
+        });
+    } else if (bannerType === 'contributions') {
+        const [, getCached] = caches.contributions[region];
+        return getCached().then(deployDate => {
+            return (
+                !engagementBannerLastClosedAt || deployDate > new Date(engagementBannerLastClosedAt)
+            );
+        });
+    }
+
+    return Promise.resolve(true);
 };
 
 // TODO - implement test selection properly
-export const selectBannerTest = (
+export const selectBannerTest = async (
     targeting: BannerTargeting,
     pageTracking: BannerPageTracking,
     baseUrl: string,
 ): Promise<BannerTestSelection | null> => {
-    return redeployedSinceLastClosed(targeting).then(hasRedeployed => {
-        if (hasRedeployed) {
-            const tests = [AusMomentContributionsBanner];
-            const testToRun = tests.find(
-                test =>
-                    targeting.alreadyVisitedCount >= test.minPageViews &&
-                    !targeting.shouldHideReaderRevenue &&
-                    !targeting.isPaidContent &&
-                    test.canRun(targeting, pageTracking),
-            );
+    const tests: BannerTest[] = [
+        AusMomentContributionsBanner,
+        DigitalSubscriptionBanner,
+        WeeklyBanner,
+    ];
+
+    for (const test of tests) {
+        if (
+            !targeting.shouldHideReaderRevenue &&
+            !targeting.isPaidContent &&
+            targeting.showSupportMessaging &&
+            targeting.alreadyVisitedCount >= test.minPageViews &&
+            test.canRun(targeting, pageTracking) &&
+            (await redeployedSinceLastClosed(targeting, test.bannerType))
+        ) {
+            const variant = test.variants[0];
```

Comment:
is it time to begin picking the variant based on the mvt value?
https://github.com/guardian/frontend/blob/15c5c493d806ea361bb67ebcb332d581f2fcb929/static/src/javascripts/projects/common/modules/experiments/ab-core.js#L52

Reply:
As discussed, I will add a TODO here - I think there was one but we accidentally deleted it!